controllers/workspace/devworkspace_controller.go

```diff
@@ -355,6 +376,9 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		}
 		reqLogger.Info("Waiting on deployment to be ready")
 		reconcileStatus.setConditionFalse(conditions.DeploymentReady, "Waiting for workspace deployment")
+		if !deploymentStatus.Requeue && deploymentStatus.Err == nil {
+			return reconcile.Result{RequeueAfter: startingWorkspaceRequeueInterval}, nil
```

Comment:
Does here the last or the first internal win?

- I mean we wait for deployment initially and scheduled reconciling in 5 secs (1);
- Deployment created pods and it invokes reconcile loop, we scheduling in 5 secs (2);
- Pod is updated several times after containers status are propagates, we scheduled N times in 5 secs(N);

The question is, how many times reconcile loops our `RequeueAfter` will initiate?

Reply:
My understanding is that reconciling due to an event cancels out a requeueAfter, but it's hard to check for sure. In my testing, I don't see bursts of "waiting on deployment" in the logs, which is what would likely happen if we were stacking requeues.